impl/src/main/java/org/ehcache/impl/serialization/CompactJavaSerializer.java

```diff
@@ -52,7 +53,7 @@
 
   private final AtomicInteger nextStreamIndex = new AtomicInteger(0);
```

Comment:
I think this is actually my bug and not yours... but since you're in here... this needs to be updated after reading the mappings from the file.